### PR TITLE
[NIS-13912]: implemented offices-api api layer

### PR DESCRIPTION
**Pull request by westcrime**

### CODE REVIEW

File: Offices/Offices.API/Controllers/DoctorsController.cs [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            return Ok(doctorDtos);
+        }
+
+        /// <summary>
+        /// Gets an doctor by ID.
+        /// </summary>
+        /// <param name="id">The ID of the doctor to retrieve.</param>
+        /// <returns>Returns the doctor object.</returns>
+        /// <response code="200">If the doctor is found</response>
+        /// <response code="204">If the doctor is not found</response>
```
Comment (Glipchik) [inline Offices/Offices.API/Controllers/DoctorsController.cs:69]: 204 for not found?

File: Offices/Offices.API/Controllers/OfficesController.cs [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+            return Ok(officeDtos);
+        }
+
+        /// <summary>
+        /// Gets an office by ID.
+        /// </summary>
+        /// <param name="id">The ID of the office to retrieve.</param>
+        /// <returns>Returns the office object.</returns>
+        /// <response code="200">If the office is found</response>
+        /// <response code="204">If the office is not found</response>
```
Comment (Glipchik) [inline Offices/Offices.API/Controllers/OfficesController.cs:73]: Same

File: Offices/Offices.API/Controllers/ReceptionistsController.cs [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+            return Ok(receptionistDtos);
+        }
+
+        /// <summary>
+        /// Gets an receptionist by ID.
+        /// </summary>
+        /// <param name="id">The ID of the receptionist to retrieve.</param>
+        /// <returns>Returns the receptionist object.</returns>
+        /// <response code="200">If the receptionist is found</response>
+        /// <response code="204">If the receptionist is not found</response>
```
Comment (Glipchik) [inline Offices/Offices.API/Controllers/ReceptionistsController.cs:69]: Same

**Review by Glipchik — CHANGES_REQUESTED** [review]

File: Offices/Offices.API/Controllers/DoctorsController.cs (hunk 1 above)
Comment (westcrime) [reply Offices/Offices.API/Controllers/DoctorsController.cs]: Fixed

File: Offices/Offices.API/Controllers/OfficesController.cs (hunk 2 above)
Comment (westcrime) [reply Offices/Offices.API/Controllers/OfficesController.cs]: Fixed

File: Offices/Offices.API/Controllers/ReceptionistsController.cs (hunk 3 above)
Comment (westcrime) [reply Offices/Offices.API/Controllers/ReceptionistsController.cs]: Fixed

File: Offices/Offices.API/Controllers/DoctorsController.cs [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+            return doctorDtos;
+        }
+
+        /// <summary>
+        /// Gets an doctor by ID.
+        /// </summary>
+        /// <param name="id">The ID of the doctor to retrieve.</param>
+        /// <returns>Returns the doctor object.</returns>
+        /// <response code="200">If the doctor is found</response>
+        /// <response code="400">If the doctor is not found</response>
```
Comment (Glipchik) [inline Offices/Offices.API/Controllers/DoctorsController.cs:69]: 404 should be returned everywhere when something wasn't found

File: Offices/Offices.API/Extensions/ApiExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+            // Adding problem details
+            services.AddProblemDetails();
+
+            // Adding validators (it will register all the validators in the same assembly in which CreateOfficeDtoValidator is defined)
+            services.AddValidatorsFromAssemblyContaining<CreateOfficeDtoValidator>();
+
+            return services;
+        }
+
+        public static void AddToModelState(this ValidationResult result, ModelStateDictionary modelState, ILogger logger)
```
Comment (Glipchik) [inline Offices/Offices.API/Extensions/ApiExtensions.cs:49]: Do you need it?

**Review by Glipchik — CHANGES_REQUESTED** [review]

File: Offices/Offices.API/Controllers/DoctorsController.cs (hunk 4 above)
Comment (westcrime) [reply Offices/Offices.API/Controllers/DoctorsController.cs:69]: fixed

File: Offices/Offices.API/Extensions/ApiExtensions.cs (hunk 5 above)
Comment (westcrime) [reply Offices/Offices.API/Extensions/ApiExtensions.cs:49]: No, I don't, fixed

**Review by Glipchik — APPROVED** [review]

**Pull request merged**